src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -0,0 +1,145 @@
+import com.amazonaws.services.ec2.model.TerminateInstancesResult;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancing;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient;
+import com.amazonaws.services.elasticloadbalancingv2.model.RegisterTargetsRequest;
+import com.amazonaws.services.elasticloadbalancingv2.model.TargetDescription;
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.models.Deployment;
+import com.conveyal.datatools.manager.models.OtpServer;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Job that is dispatched during a {@link DeployJob} that spins up EC2 instances. This handles waiting for the server to
+ * come online and for the OTP application/API to become available.
+ */
+public class MonitorServerStatusJob extends MonitorableJob {
+    private static final Logger LOG = LoggerFactory.getLogger(MonitorServerStatusJob.class);
+    private final Deployment deployment;
+    private final Instance instance;
+    private final OtpServer otpServer;
+    private final AmazonEC2 ec2 = AmazonEC2Client.builder().build();
+    private final CloseableHttpClient httpClient = HttpClients.createDefault();
+    // If the job takes longer than XX seconds, fail the job.
+    private static final int TIMEOUT_MILLIS = 60 * 60 * 1000; // One hour
+
+    public MonitorServerStatusJob(String owner, Deployment deployment, Instance instance, OtpServer otpServer) {
+        super(
+            owner,
+            String.format("Monitor server setup %s", instance.getPublicIpAddress()),
+            JobType.MONITOR_SERVER_STATUS
+        );
+        this.deployment = deployment;
+        this.instance = instance;
+        this.otpServer = otpServer;
+        status.message = "Checking server status...";
+    }
+
+    @Override
+    public void jobLogic() {
+        long startTime = System.currentTimeMillis();
+        // Get OTP URL for instance to check for availability.
+        String otpUrl = String.format("http://%s/otp", instance.getPublicIpAddress());
+        boolean otpIsRunning = false, routerIsAvailable = false;
+        // Progressively check status of OTP server
+        if (deployment.buildGraphOnly) {
+            // FIXME No need to check that OTP is running. Just check to see that the graph is built.
+//            FeedStore.s3Client.doesObjectExist(otpServer.s3Bucket, );
+        }
+        // First, check that OTP has started up.
+        status.update("Instance status is OK. Waiting for OTP...", 20);
+        while (!otpIsRunning) {
```

Comment:
As noted in https://github.com/ibi-group/datatools-server/pull/225/files#r324391156, there are many things that occur before OTP starts. These items could be reported to better update progress, or if an fatal error occured, the job could be immediately failed instead of waiting for the timeout.